lib/mongoid/association/nested/many.rb

```diff
@@ -45,6 +45,7 @@ def build(parent, options = {})
         # @param [ Hash ] attributes The attributes hash to attempt to set.
         # @param [ Hash ] options The options defined.
         def initialize(association, attributes, options = {})
+          attributes = attributes&.to_h if attributes.respond_to?(:keys)
```

Comment:
What is the purpose of  `&` on this line?

Reply:
I had initially assumed this could be nil, so .to_h is only called if attributes isn't nil. It's not need after I added the :keys test .